### PR TITLE
Add find or create method to HubSpot Service

### PR DESCRIPTION
**Pull request by tochman**
PT Story: https://www.pivotaltracker.com/story/show/178850941

```
As a system user
In order to avoid errors when exporting data to HubSpot
I want to be able to check if a user with a specific email already exist on HubSpot
```

* adds find or create method to HubSpot service
* refactors specs for inquiry create and HS export
* renames HubSpot service move -> send

### CODE REVIEW

**Comment by tochman** [comment]
This is still WIP. Must revisit some of the test cases.

**Review by 1CIM — APPROVED** [review]
Can't really see any problem except that i need to look att the video you made.
Do you remember if you sent it to us or twitch or youtube? 😅

File: app/services/hub_spot_service.rb [hunk 1]
```diff
@@ -14,16 +14,33 @@ def self.move(inquiry)
...
+        RestClient.get("https://api.hubapi.com/contacts/v1/contact/email/#{inquiry.email}/profile?hapikey=#{@api_key}")
+      ).symbolize_keys
+    rescue StandardError => e
+      contact = JSON.parse(e.response.body).symbolize_keys
+    end
+    if contact[:status] != 'error'
+      RestClient.post(
+        "https://api.hubapi.com/contacts/v1/contact/vid/#{contact[:vid]}/profile?hapikey=#{@api_key}",
+        data.to_json, { content_type: :json, accept: :json }
+      )
```
Comment (offcast-dev) [inline app/services/hub_spot_service.rb:30]: So this request will happen if the contact is found and should add new note to it, correct? However I don't see you testing that case, only new fixture is that contact is not found, which is `else` in this if statement.

**Review by offcast-dev — COMMENTED** [review]
Need some clarification on testing coverage.

File: app/services/hub_spot_service.rb (hunk 1 above)
Comment (tochman) [reply app/services/hub_spot_service.rb:30]: Yeah... I'm a bit lost in this ATM. Should probably not code complicated features when tired...

**Review by offcast-dev — APPROVED** [review]

**Pull request merged**